Fix #25

katterfelto opened the pull request:
This fixes the closed issue #25 which is causing me problems on a customers system and I have managed to reproduce on a test system.

I have tested it and the exception is still being thrown but is now handled silently.

katterfelto committed:
Add catch round GDI+ call

lextm commented:
Since you have an environment to reproduce this issue, can you print out the exact `ClientRectangle` information when the exception occurs? What is the width and height that triggers the exception? That might give us a better fix (test against such parameters) instead of blindly catching all exceptions.

lextm commented:
If possible, can you also share information on the details of your testing environment? What kind of setup is required?

DaisyDave commented:
I am happy to provide more information. I added the catch because to my eyes there didn't seem to be any problem with the data.

For us the problem only seems to occur on PC's running 64bit Windows 2008 R2 Standard. As far as I know we are only ever using Remote Desktop to access them. I should be able to get a screen, keyboard and mouse connected to my local test environment if required. Our application is a 32bit process. The problem occurs when we are closing all the windows in the workspace prior to unloading the configuration from our application. Of the 2 servers on our customers site one works OK whilst the othe regularly throws this exception, the only difference between the 2 is the configuration loaded into the application.

I am testing by running a release version of the application on the server and using Visual Studio 2017 Remote Debugging Tools to attach to the process from another PC. In my local test environment the problem occurs even when I do not have the remot debugger attached.

I am attaching 2 screen shots showing the ClientRectangle and e.Graphics data. At the moment nobody is asking to use the server for other tests so I should be able to run further the tests. Let me know what more you need.
![clientrectangle](https://user-images.githubusercontent.com/3041671/26851714-962b5eee-4b03-11e7-80a2-fae26a1fb044.png)
![e graphics](https://user-images.githubusercontent.com/3041671/26851715-9646b43c-4b03-11e7-844d-d204314916db.png)

roken commented:
Swallowing exceptions has a very strong smell.

lextm commented:
@DaisyDave Thanks. Sounds like it is more related to resource usage, instead of rectangle size. Can you follow Hans Passant's suggestion in https://stackoverflow.com/questions/9362184/how-to-resolve-the-winforms-error-a-generic-error-occurred-in-gdi to check if there is an obvious leak?

bvogelcvs committed:
Bugfix Issue #488 - VisibleNestedPaneCollection needs to look for PreviousDisplayingPane instead of PreviousPane

lextm committed:
Simplified NuGet definition files.

lextm committed:
Added the latest patch under patch controller.

lextm committed:
Bumped the version number to 2.15.

lextm committed:
Create CODE_OF_CONDUCT.md

joelbyren committed:
HitTestCaption in Mono

joelbyren committed:
Fix for #494

joelbyren committed:
Brackets to have else on correct if

joelbyren committed:
* StyleCop rule

joelbyren committed:
* StyleCop rule

joelbyren committed:
Update Resources.resx

jotatsu committed:
default text color to theme

default text color to theme, if not it will be always black (dark themes conflict for some items that are not in the if-else scheme).

lextm committed:
Upgraded to VS2017. Changed default framework version.

lextm committed:
Completely removed VS2010 garbage.

lextm committed:
Removed the remaining bits.

lextm committed:
Bumped the version.

lextm committed:
Added a patch for #502 to avoid wrong active control assignment.

lextm committed:
Fixed .NET Framework 3.5 compilation.

lextm committed:
Bumped package version.

lextm committed:
Revised NuGet package publishing script.

lextm committed:
Bumped the version to 3.0.0.

katterfelto committed:
Merge branch 'master' of https://github.com/DaisyDave/dockpanelsuite

Krid82 commented:
I also ran into this problem serveral times when using Remote Desktop to connect to a Window 7 PC from a Windows 10 PC. I cannot reproduce it though, it seems to be completely random. It also happens with version 3.0.2. 

I observed the number of handles of the application, as suggested by Hans Passant on stackoverflow. I not could find anything suspicious though. It uses about 600 handles and that number remains more or less constant. When I ignore the expection by clicking continue, a red frame is drawn around the docking area. That frame stays there until I close the application.

When looking at DaisyDaves's screenshots, it seems to me that the problem is the extreme size of the clipbounds of the graphics object. What units does the ClipBounds structure use? Is it pixels or something else?

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/dockpanelsuite/dockpanelsuite?pullRequest=487) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/dockpanelsuite/dockpanelsuite?pullRequest=487) before we can accept your contribution.<br/>**2** out of **5** committers have signed the CLA.<br/><br/>:white_check_mark: katterfelto<br/>:white_check_mark: joelbyren<br/>:x: lextm<br/>:x: bvogelcvs<br/>:x: jotatsu<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/dockpanelsuite/dockpanelsuite?pullRequest=487) it.</sub>

tajbender reviewed (commented):
looks good
